misk-api/src/main/kotlin/misk/api/RequestContext.kt

```diff
@@ -0,0 +1,16 @@
+package misk.api
+
+import misk.web.DispatchMechanism
+import okhttp3.HttpUrl
+
+interface RequestContext {
+  val url: HttpUrl
+  val dispatchMechanism: DispatchMechanism
+  val requestHeadersContext: RequestHeadersContext
+}
+
+interface RequestHeadersContext {
```

Comment:
This is to avoid having to create OKHttp headers out of Armeria headers. This interface is very easy to implement for both misk and Armeria.

Reply:
If the `misk-api` project is already going to pull in OkHttp as an `api` dependency, I wonder if it'd be easier to just have an extension function that converts Armeria headers into OkHttp headers?

Adding new types everywhere really worries me about the long term maintainability of this effort, and in some cases seems a bit unnecessary.